upload.go

```diff
@@ -95,29 +98,22 @@ func (w *WoClient) Upload2C(spaceType string, file Upload2CFile, targetDirId str
 		}
 		formData["partSize"] = strconv.FormatInt(partSize, 10)
 		formData["partIndex"] = strconv.FormatInt(partIndex, 10)
-
-		req := w.NewRequest().
-			SetResult(&resp).
-			ForceContentType("application/json;charset=UTF-8").
-			SetHeaders(map[string]string{
-				"Origin":     "https://pan.wo.cn",
-				"Referer":    "https://pan.wo.cn/",
-				"User-Agent": w.ua,
-			}).
-			SetMultipartFormData(formData).
-			SetMultipartField("file", file.Name, file.ContentType, io.LimitReader(file.Content, partSize))
-		if opt.Ctx != nil {
-			req.SetContext(opt.Ctx)
-		}
-		res, err := req.Post(uploadURL)
+		err := w.uploadPart(opt.Ctx, uploadURL, file, formData, partIndex, partSize, &resp)
 		if err != nil {
-			return "", err
-		}
-		if res.IsError() {
-			return "", fmt.Errorf("partIndex: %d, failed to upload2C with http status: %d, body: %s", partIndex, res.StatusCode(), res.String())
-		}
-		if resp.Code != "0000" {
-			return "", fmt.Errorf("partIndex: %d, failed to upload2C with code: %s, msg: %s", partIndex, resp.Code, resp.Msg)
+			for i := 0; i < opt.RetryTimes; i++ {
+				_, serr := file.Content.Seek(finishedSize, 0)
+				if serr != nil {
+					return "", serr
+				}
+				err = w.uploadPart(opt.Ctx, uploadURL, file, formData, partIndex, partSize, &resp)
+				if err == nil {
+					break
+				}
+				log.Printf("retry upload partIndex: %d, err: %v\n", partIndex, err)
```

Comment:
There's no need to print errors at this level; they should be managed by higher-level applications.